libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using prompts = Microsoft.Bot.Builder.Prompts;
+using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
+using Microsoft.Bot.Schema;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Prompts;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class OAuthPromptSettingsWithTimeout : prompts.OAuthPromptSettings
+    {
+        public int? Timeout { get; set; }
+    }
+
+    public class OAuthPromptState : PromptOptions
+    {
+        public OAuthPromptState(PromptOptions defaultPromptOptions) : base()
+        {
+            if (defaultPromptOptions != null)
+            {
+                PromptString = PromptString ?? defaultPromptOptions.PromptString;
+                PromptActivity = PromptActivity ?? defaultPromptOptions.PromptActivity;
+                Speak = Speak ?? defaultPromptOptions.Speak;
+                RetryPromptString = RetryPromptString ?? defaultPromptOptions.RetryPromptString;
+                RetryPromptActivity = RetryPromptActivity ?? defaultPromptOptions.RetryPromptActivity;
+                RetrySpeak = RetrySpeak ?? defaultPromptOptions.RetrySpeak;
+            }
+        }
+
+        public DateTime Expires { get; set; }
+    }
+
+    public class OAuthPrompt : Control, IDialog, IDialogContinue
+    {
+        private prompts.OAuthPrompt _prompt;
+        private OAuthPromptSettingsWithTimeout _settings;
+
+        public OAuthPrompt(OAuthPromptSettingsWithTimeout settings, PromptValidator<TokenResult> validator = null)
+        {
+            _settings = settings ?? throw new ArgumentException(nameof(settings));
+            _prompt = new prompts.OAuthPrompt(settings, validator);
+        }
+
+        public async Task DialogBegin(DialogContext dc, object dialogArgs = null)
+        {
+            if (dc == null)
+                throw new ArgumentNullException(nameof(dc));
+            if (dialogArgs == null)
+                throw new ArgumentNullException(nameof(dialogArgs));
+
+            var promptOptions = (PromptOptions)dialogArgs;
+
+            //persist options and state
+            var timeout = _settings.Timeout.HasValue ? _settings.Timeout.Value : 54000000;
```

Comment:
Where does this magic number come from? 

54,000,000? That's a weird number. 

First, remove this to a constant and add detailed comments around it. 
Better yet, see if you can remove it and use a shared common behavior taken from a base prompt..

Reply:
removed to a constant and added a comment.

---
In reply to: [184819087](https://github.com/Microsoft/botbuilder-dotnet/pull/520#discussion_r184819087) [](ancestors = 184819087)